### PR TITLE
PDE-3822 feat(cli): Only require fields for private integrations via CLI

### PR DESCRIPTION
**Pull request by gregilo**
Currently, app meta fields/settings are only editable for private apps in the visual builder. Now that the CLI allows users to update these settings via CLI, we need to ensure that they can only update these settings for **private** apps.

[Jira story](https://zapierorg.atlassian.net/browse/PDE-3822)

### CODE REVIEW

File: packages/cli/src/oclif/commands/register.js [hunk 1]
```diff
@@ -132,6 +132,13 @@ class RegisterCommand extends ZapierBaseCommand {
       this.startSpinner('Retrieving details for your integration');
       this.app = await getWritableApp();
       this.stopSpinner();
+
+      // Block non-private apps from updating settings
+      if (this.app?.intention !== 'private') {
```
Comment (eliangcs) [inline packages/cli/src/oclif/commands/register.js:137]: Should we use the `status` field instead?

```suggestion
      if (this.app?.status !== 'private') {
```

Comment (gregilo) [reply packages/cli/src/oclif/commands/register.js]: @eliangcs Great catch - I was unaware of how `app.status` is being used. I checked [the logic we're using in VB](https://gitlab.com/zapier/team-developer-platform/dev-platform/-/blob/9fa28d8bacd04ebdad5937bd039c71aede4ede47/web/frontend/assets/app/entities/CliApp/CliApp.ts#L96) to limit users from updating app settings for published apps and decided to implement that logic in 576b463ae49a1007ce4014c28d62bb36e26adbfd.

Can you please re-review?

Comment (eliangcs) [reply packages/cli/src/oclif/commands/register.js]: Thanks for checking the logic on the UI side! Looks good to me!

**Review by eliangcs — APPROVED** [review]

File: packages/cli/src/oclif/commands/register.js (hunk 1 above)
Comment (gregilo) [reply packages/cli/src/oclif/commands/register.js]: @eliangcs Thanks for reviewing! I saw that you pushed ba6ca25072f70d22c138e8e7d60e8ab9a26f2bbf, but decided to revert it - is there a more ideal solution than what I've pushed?

Comment (eliangcs) [reply packages/cli/src/oclif/commands/register.js]: @gregilo I thought the `public_ish` field returned by the server is defined as `status in ('public', 'beta')` and we could use it, but then I found out `public_ish` is in fact defined as `status == 'public'`, so I reverted it.

**Pull request merged**